pkg/planner/cascades/memo/group_expr.go

```diff
@@ -15,9 +15,8 @@
 package memo
 
 import (
-	"io"
-
 	base2 "github.com/pingcap/tidb/pkg/planner/cascades/base"
+	"github.com/pingcap/tidb/pkg/planner/cascades/util"
 	"github.com/pingcap/tidb/pkg/planner/core/base"
 	"github.com/pingcap/tidb/pkg/planner/pattern"
```

Comment:
Maybe we need to organize the old cascades struct into new one

Reply:
yes, we did, currently all old stuff has been moved from ./cascades into ./cascades/old pkg, cause we want to use cascades pkg

Reply:
![image](https://github.com/user-attachments/assets/230f8abb-799b-4765-87c1-3877d498b60c)